src/main/java/apoc/atomic/Atomic.java

```diff
@@ -0,0 +1,213 @@
+			Lock lock = tx.acquireReadLock(propertyContainer);
+			List<Object> values = insertValueIntoArray(oldValue, position, value);
+			Class clazz = Class.forName(values.toArray()[0].getClass().getName());
+			newValue = Array.newInstance(clazz, values.size());
+			try {
+				System.arraycopy(values.toArray(), 0, newValue, 0, values.size());
+			} catch (Exception e) {
+				String message = "Property's array value has type: " + values.toArray()[0].getClass().getName() + ", and your value to insert has type: " + value.getClass().getName();
+				throw new ArrayStoreException(message);
+			}
+			propertyContainer.setProperty(property, newValue);
+			tx.success();
+			lock.release();
+		}
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * remove a value into an array property value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.remove(node/relatonship,propertyName,position) remove the element at position 'position'")
+	public Stream<AtomicResults> remove(@Name("container") Object container, @Name("propertyName") String property, @Name("position") Long position) throws ClassNotFoundException {
+		checkIsPropertyContainer(container);
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		checkPropertyExists(propertyContainer,property);
+		Object oldValue = propertyContainer.getProperty(property);
+		Object newValue;
+		try(Transaction tx = db.beginTx()){
+			Lock lock = tx.acquireReadLock(propertyContainer);
+			Object[] arrayBackedList = new ArrayBackedList(oldValue).toArray();
+			List<Object> value = new ArrayList<>();
+			for(int i= 0; i< arrayBackedList.length; i++) {
+				if(i != position.intValue()) value.add(arrayBackedList[i]);
+			}
+			Class clazz = Class.forName(value.toArray()[0].getClass().getName());
+			newValue = Array.newInstance(clazz, value.size());
+			System.arraycopy(value.toArray(), 0, newValue, 0, value.size());
+			propertyContainer.setProperty(property, newValue);
+			tx.success();
+			lock.release();
+		}
+		return Stream.of(new AtomicResults(oldValue,newValue));
+	}
+
+	/**
+	 * update the property's value
+	 */
+	@Procedure(mode = Mode.WRITE)
+	@Description("apoc.atomic.update(node/relatonship,propertyName,updateOperation) update a property's value with a cypher operation (ex. \"n.prop1+n.prop2\")")
+	public Stream<AtomicResults> update(@Name("container") Object container, @Name("propertyName") String property, @Name("operation") String operation){
+		checkIsPropertyContainer(container);
+		PropertyContainer propertyContainer = (PropertyContainer) container;
+		checkPropertyExists(propertyContainer,property);
+		Object oldValue = propertyContainer.getProperty(property);
+		try(Transaction tx = db.beginTx()){
+			Lock lock = tx.acquireReadLock(propertyContainer);
+			String cypherStatement = null;
+			if(container instanceof Node)
+				cypherStatement= "start n=node("+ ((Node) propertyContainer).getId() + ") set n."+property+"="+operation;
```

Comment:
just pass the propertyContainer instance to db.execute as parameter, e.g. "this" or "n" or "_" and set the value